src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -177,6 +273,17 @@ static int fstatat64_wrapper(int dfd, const char *path,
 }
 #endif
 
+#if defined(__linux__)
+static int statx_wrapper(int dirfd, const char *restrict pathname, int flags,
+                         unsigned int mask, struct my_statx *restrict statxbuf,
+                         int follow_symlink) {
+    if (follow_symlink == NO_FOLLOW_SYMLINK) {
+      flags |= AT_SYMLINK_NOFOLLOW;
```

Comment:
I think it would be better if statx_wrapper had the same signature as statx. Have you tried adding the AT_SYMLINK_NOFOLLOW flag at the use-site?

Reply:
Sure, I can do that.